internal/controller/pkg/resolver/reconciler.go

```diff
@@ -375,3 +401,95 @@ func (r *Reconciler) findDependencyVersion(ctx context.Context, dep *v1beta1.Dep
 
 	return addVer, nil
 }
+
+// UpdatableVersionFinder is an implementation of versionFinder that allows for version upgrade/downgrade capability considering parent constraints.
+type UpdatableVersionFinder struct {
+	fetcher xpkg.Fetcher
+	config  xpkg.ConfigStore
+}
+
+// FindValidDependencyVersion finds a valid version with version upgrade/downgrade capability considering parent constraints.
```

Comment:
```suggestion
// FindValidDependencyVersion finds a valid version with version upgrade capability considering parent constraints.
```

Dropping downgrade since we won't support it.